app/src/main/java/ch/epfl/culturequest/authentication/Authenticator.java

```diff
@@ -89,9 +91,21 @@ public static CompletableFuture<String> signIn(ComponentActivity activity) {
         } else {
                 Database.getProfile(getCurrentUser().getUid()).handle((profile, throwable) -> {
                     if (profile != null) {
-                        Profile.setActiveProfile(profile);
-                        AndroidUtils.redirectToActivity(activity, NavigationActivity.class);
-                        future.complete("User signed in with an existing profile");
+                        // the following part is used to check at every login if the user has a
+                        // new device token
+                        List<String> deviceTokens = profile.getDeviceTokens();
+                        FireMessaging.getDeviceToken().whenComplete((token, throwable1) -> {
+                            // if the current device token is not already in the list, add it
+                            // and update the database
+                            if (!deviceTokens.contains(token)) {
+                                deviceTokens.add(token);
+                                profile.setDeviceTokens(deviceTokens);
+                                Database.setDeviceTokens(profile.getUid(), deviceTokens);
+                            }
```

Comment:
In `ProfileCreatorActivity.java`, you first checked if the throwable was null, and then added the token anyway. I think you should be consistent in how you do things.

Reply:
ok I will add a check on the throwable too here